src/components/Extensions/ExtensionItem.tsx

```diff
@@ -0,0 +1,42 @@
+'use client'
+import { ExtensionType } from "@/types/extensionsTypes";
+import { Verified } from "lucide-react";
+import Link from "next/link";
+import { usePathname } from "next/navigation";
+
+export function ExtensionItem({ extension }: { extension: ExtensionType }) {
+  const pathname = usePathname();
+
+  return (
+    <Link
+      href={`/extensions/${extension.extensionId}`}
+      data-active={pathname?.endsWith(extension.extensionId)}
+      className="flex w-full gap-4 py-2 pl-4 data-[active=true]:bg-[#2a273f] "
+    >
+      <div className="flex items-center w-[42px]">
+        {/* eslint-disable-next-line @next/next/no-img-element */}
+        <img
+          src={
+            extension.versions[0].files.find(
+              (file) =>
+                file.assetType ===
+                "Microsoft.VisualStudio.Services.Icons.Default"
+            )?.source ?? ""
+          }
+          alt={extension.extensionName}
+          className="min-w-[42px] h-[42px]"
+        />
+      </div>
+      <div className="flex flex-col leading-5 text-sm text-[#908caa] max-w-full w-[calc(256px-42px-16px-24px)]"> {/* calc = 100% parent width (256) - img width (42) - parent padding left (16) - padding from scroll (24) */}
```

Comment:
I don't think this calc is necessary while using flex or grid 🤔

Reply:
This is with the calc
![image](https://user-images.githubusercontent.com/31166233/224522020-093332c5-efa4-494f-aa43-48fcc3f80367.png)

This is without
![image](https://user-images.githubusercontent.com/31166233/224522039-8081f0fb-4636-4129-ac0e-21afc140511c.png)

basically the div element doesn't respect the parents width, and text oveflow "ellipsis" will not show because its hidden behind the content panel, if you have suggestions on how to make this div respect it's parent width, I will happily make the change, I also didn't like the calc as it's confusing, but it was the only way to mimic visual code style, and to be a little bit less confusing I added the comment